addons/pos_restaurant/static/src/js/ChromeWidgets/TicketButton.js

```diff
@@ -7,6 +7,29 @@ odoo.define('pos_restaurant.TicketButton', function (require) {
 
     const PosResTicketButton = (TicketButton) =>
         class extends TicketButton {
+            async onClick() {
+                if (this.env.pos.config.iface_floorplan && !this.props.isTicketScreenShown && !this.env.pos.table) {
+                    await this._syncAllFromServer();
+                    this.showScreen('TicketScreen');
+                } else {
+                    super.onClick();
+                }
+            }
+            async _syncAllFromServer() {
+                const pos = this.env.pos;
+                try {
+                    for (const floor of pos.floors) {
+                        for (const table of floor.tables) {
+                            await pos.replace_table_orders_from_server(table);
```

Comment:
this introduce a performance issue

Takes too long to load tickets screen if your pos has several floors and several tables on each


Also spams the method get_table_draft_orders, should be refactor to make just one call per floor

Reply:
It was decided due to the stable policy to not add something that already exists unless there's an explicit issue about it.
In v16 and master, this PR is different to tackle on the performance you mentioned